### PR TITLE
Add cherry-pick action for release branches

### PR DESCRIPTION
**Pull request by ggriffiths**
Signed-off-by: Grant Griffiths <ggriffiths@purestorage.com>
**What this PR does / why we need it**:
Adds a github action to automate cherry-picking to release branches. You can add a job for your feature branch if you want as well.

Steps:
1. Open PR against openstorage master
2. Label your PR with a the correct cherry-pick label: `cherry-pick/release-9.4` or `cherry-pick/release-9.3`
3. Get approval & merge
4. PR will be opened automatically against the release branch for you.

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: .github/workflows/on-pr-closed.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+on:
+  pull_request:
+    branches:
+      - master
+    types: ["closed"]
```
Comment (dahuang-purestorage) [inline .github/workflows/on-pr-closed.yml:5]: will this be called if the pr was closed but not merged? 
should we try something mentioned in https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-when-a-pull-request-merges-1

Comment (ggriffiths) [reply .github/workflows/on-pr-closed.yml:5]: Ah good catch, yes let's do that

**Review by dahuang-purestorage — APPROVED** [review]
lgtm

File: .github/workflows/on-pr-closed.yml (hunk 1 above)
Comment (ggriffiths) [reply .github/workflows/on-pr-closed.yml:5]: Ok updated the PR @dahuang-purestorage. Tested it here too:
https://github.com/ggriffiths/openstorage/pull/18

First, the action was skipped when it closed
Second, it ran when I reopened and merged the PR.

**Pull request merged**